src/test/java/org/elasticsearch/index/analysis/SkroutzGreekStemmerTest.java

```diff
@@ -56,23 +56,25 @@ public void testSkroutzGreekStemmerExceptionsZero()
     Field field = stemmer.getClass().getDeclaredField("stepZeroExceptions");
     field.setAccessible(true);
 
-    Map<String, char[]> stepZeroExceptions = (Map) field.get(stemmer);
-
-   for (Map.Entry<String, char[]> entry : stepZeroExceptions.entrySet()) {
-     Assert.assertTrue(entry.getKey().length() >= entry.getValue().length,
-             "Make sure that the stem is shorter/equal than the original.");
-   }
+    CharArrayMap stepZeroExceptions = (CharArrayMap) field.get(stemmer);
+    CharArrayMap.EntryIterator it = stepZeroExceptions.entrySet().iterator();
+    while(it.hasNext()) {
```

Comment:
Yes. Not very intuitive right? That was my original assumption and confirmed with debugging and passing tests.

https://lucene.apache.org/core/5_1_0/analyzers-common/org/apache/lucene/analysis/util/CharArrayMap.EntryIterator.html#currentValue()